pytorch_lightning/core/lightning.py

```diff
@@ -1094,7 +1095,9 @@ def training_step(...):
         self._verify_is_manual_optimization('manual_backward')
 
         # backward
+        self._running_manual_optim = True
```

Comment:
Would it be easier to add an optional arg to `backward`? or do we need the state `_running_manual_optim` somewhere else as well?

Reply:
Great question ! 
I thought about it, but it would mean broadcasting a new parameter into training_loop, accelerator, etc... 
I though the simpler, the better. And now, it defines a scope around manual_optimization if we need it. 
Maybe an actual context manager would be cleaner there.
 @SeanNaren @Borda Any thoughts ?

Reply:
I could reproduce the bug with ddp but not locally. Need more investigation.

Reply:
Yes am confused why this bug only appears in ddp. Could you confirm you don't see the behaviour with ddp_cpu?

Reply:
Need more investigation :)